assets/app/scripts/controllers/builds.js

```diff
@@ -44,7 +44,43 @@ angular.module('openshiftConsole')
     watches.push(DataService.watch("buildConfigs", $scope, function(buildConfigs) {
       $scope.buildConfigs = buildConfigs.by("metadata.name");
       Logger.log("buildConfigs (subscribe)", $scope.buildConfigs);
-    }));    
+    }));
+
+    var associateRunningBuildToBuildConfig = function(buildsByBuildConfig) {
+      var buildConfigBuildsInProgress = {}; 
+      for (var buildConfigName in buildsByBuildConfig) {
+        var buildConfigBuilds = buildsByBuildConfig[buildConfigName];
+        buildConfigBuildsInProgress[buildConfigName] = {};
+        for (var buildName in buildConfigBuilds) {
+          var build = buildConfigBuilds[buildName];
+          var buildStatus = build.status;
+          if (buildStatus === "New" || buildStatus === "Pending" || buildStatus === "Running") {
+            buildConfigBuildsInProgress[buildConfigName][buildName] = build;
+          }
+        }
+      }
+      return buildConfigBuildsInProgress;
+    }
+
+    watches.push(DataService.watch("builds", $scope, function(builds, action, build){
+      if (build) {
+        var buildConfigName = build.metadata.labels.buildconfig;
+        var buildName = build.metadata.name;
+      }
+      if (!action) {
+        // Loading of the page that will create buildConfigBuildsInProgress structure, which will associate running build to his buildConfig.
+        $scope.buildConfigBuildsInProgress = associateRunningBuildToBuildConfig($scope.buildsByBuildConfig);
+      } else if (action === 'ADDED'){
+        // When new build id instantiated/cloned associate him to his buildConfig and add him into buildConfigBuildsInProgress structure.
+        $scope.buildConfigBuildsInProgress[buildConfigName][buildName] = build;
```

Comment:
Actually this may not be an issue since you were using buildsByBuildConfig to set things up and that map is generated just based on the builds and their labels.